src/index.tsx

```diff
@@ -795,7 +794,7 @@ export class UnControlled extends React.Component<IUnControlledCodeMirror, any>
     let update = true;
 
     if (SERVER_RENDERED) update = false;
-    if (this.detached) update = false;
+    if (this.detached && nextProps.detach) update = false;
```

Comment:
This is the only change outside of `componentWillReceiveProps` / `componentDidUpdate`. This is necessary (tests fail otherwise) because `shouldComponentUpdate` runs _after_ `componentWillReceiveProps` but _before_ `componentDidUpdate`. Without this change, once we were in a detached state, we would ignore any subsequent attempts via prop changes to re-attach.